spacy/cli/debug_data.py

```diff
@@ -247,6 +253,56 @@ def debug_data(
                     msg.warn(f"No examples for texts WITHOUT new label '{label}'")
                     has_no_neg_warning = True
 
+            with msg.loading("Obtaining span characteristics..."):
+                span_characteristics = _get_span_characteristics(
+                    train_dataset, gold_train_data, spans_key
+                )
+
+            msg.info(f"Span characteristics for spans_key `{spans_key}`")
+            msg.info("SD = Span Distinctiveness, BD = Boundary Distinctiveness")
+            _print_span_characteristics(span_characteristics)
+
+            msg.info(
+                "It's recommended to use a maximum of "
+                f"{math.ceil(span_characteristics['avg_length'])} ngrams "
+                "when you're using the n-gram suggester function "
+                f"(avg={round(span_characteristics['avg_length'],2)}, "
+                f"min={span_characteristics['min_length']}, max={span_characteristics['max_length']})",
+                show=verbose,
+            )
```

Comment:
I think this suggestion should be based on recall rather than average length and analyze both min and max. (It's not too hard to construct a case with outliers where the average length leads to <20% recall.)

I think it might make sense to present several min/max settings with recall (something like 75%, 85%, 95%) and provide info/hints about the speed/RAM tradeoffs involved so the users can make a decision themselves.

Reply:
Hi @adrianeboyd , thanks for the comment! I'm curious how I can best implement this. Do you mean something like:

```
If you want to get:
75% recall, then set min=$A, max=$B,
85% recall, then set min=$C, max=$D,
95% recall, then set min=$E, max=$F
```

where `$A` to `$F` are some integers. I think this might fall under the  `recommend` command? 

Also if that's the case, perhaps I should remove the highlighted line, `It's recommended to use a maximum of ...`, from `debug data`, and insert it to `recommend` instead? 🤔

Reply:
I'm not sure what you mean by "insert to `recommend`"?

I don't know enough about algorithms that would make this easy/fast to do for the general case. Maybe instead of a direct recommendation, you can just present the percentage of spans that fall in each ngram length and point out that there are speed/RAM tradeoffs in including all outliers?

Reply:
> I'm not sure what you mean by "insert to recommend"?

It's a "new" command that we're considering to include more explicit/direct config recommendations to the user based on their dataset and other factors. 

> Maybe instead of a direct recommendation, you can just present the percentage of spans that fall in each ngram length and point out that there are speed/RAM tradeoffs in including all outliers?

Ah that's doable! I'll update the PR to include this :)

Reply:
I updated the PR here to improve span length reporting: https://github.com/explosion/spaCy/pull/10668/commits/d0b7db759c53a3f1e37971f00e272bb7666bfc91

![image](https://user-images.githubusercontent.com/12949683/167062258-66e10308-06ba-4006-a237-f698b1f27a7c.png)

Reply:
I think presenting the n-gram lengths as strings and in the order is kind of confusing. I think it's still easy enough to understand if it's sorted in the order of n-gram lengths?

Reply:
Ah sure that makes a lot of sense :) Will update the PR

Reply:
Updated in: https://github.com/explosion/spaCy/pull/10668/commits/1e796fb8e6532846b19163c1175c1128347be41b